vllm/model_executor/layers/fused_moe/fused_moe.py

```diff
@@ -326,27 +328,39 @@ def fused_topk(
                                         dtype=torch.float32)
         topk_weights, topk_ids = torch.topk(routing_weights, topk, dim=-1)
     else:
-        import vllm._moe_C as moe_kernels
-
-        topk_weights = torch.empty(M,
-                                   topk,
-                                   dtype=torch.float32,
-                                   device=hidden_states.device)
-        topk_ids = torch.empty(M,
-                               topk,
-                               dtype=torch.int32,
-                               device=hidden_states.device)
-        token_expert_indicies = torch.empty(M,
-                                            topk,
-                                            dtype=torch.int32,
-                                            device=hidden_states.device)
-        moe_kernels.topk_softmax(
-            topk_weights,
-            topk_ids,
-            token_expert_indicies,
-            gating_output.float(),  # TODO(woosuk): Optimize this.
-        )
-        del token_expert_indicies  # Not used. Will be used in the future.
+        if num_expert_group == 0:
+            import vllm._moe_C as moe_kernels
+
+            topk_weights = torch.empty(M,
+                                    topk,
+                                    dtype=torch.float32,
+                                    device=hidden_states.device)
+            topk_ids = torch.empty(M,
+                                topk,
+                                dtype=torch.int32,
+                                device=hidden_states.device)
+            token_expert_indicies = torch.empty(M,
+                                                topk,
+                                                dtype=torch.int32,
+                                                device=hidden_states.device)
+            moe_kernels.topk_softmax(
+                topk_weights,
+                topk_ids,
+                token_expert_indicies,
+                gating_output.float(),  # TODO(woosuk): Optimize this.
+            )
+            del token_expert_indicies  # Not used. Will be used in the future.
+        else:
+            scores = torch.softmax(gating_output, dim = -1)
```

Comment:
Since these don't really share anything with the rest of the function, to avoid too many if conditions here, it will be best to make a new function
```
def fused_grouped_topk(
    hidden_states: torch.Tensor,
    gating_output: torch.Tensor,
    topk: int,
    renormalize: bool,
    num_expert_group: int = 0,
    topk_group: int  = 0,
)
```

and then call the `fused_grouped_topk` and `fused_experts` functions separately like in https://github.com/vllm-project/vllm/pull/4652/files#diff-7f2897118411db2b0196a83729e1ae3dedb5a450b3abcb14020d9c6cdd716ad3R170